pyxem/tests/conftest.py

```diff
@@ -194,3 +201,55 @@ def vector_library():
     lattice = diffpy.structure.Lattice(1, 1, 1, 90, 90, 90)
     library.structures = [diffpy.structure.Structure(lattice=lattice)]
     return library
+
+
+@pytest.fixture
+def test_patterns():
+    patterns = hs.signals.Signal2D(np.zeros((3, 1, 128, 128)))
+    _001_indexs = (
+        np.array([31, 31, 31, 64, 64, 97, 97, 97]),
+        np.array([31, 64, 97, 31, 97, 31, 64, 97]),
+    )
+    _101_indexs = (
+        np.array([17, 17, 17, 64, 64, 111, 111, 111]),
+        np.array([31, 64, 97, 31, 97, 31, 64, 97]),
+    )
+    _111_indexs = (
+        np.array([23, 23, 64, 64, 105, 105]),
+        np.array([40, 88, 17, 111, 40, 88]),
+    )
+    for i in range(8):
+        patterns.inav[0, 0].isig[_001_indexs[1][i], _001_indexs[0][i]] = 24.0
+        patterns.inav[0, 1].isig[_101_indexs[1][i], _101_indexs[0][i]] = 24.0
+    for i in range(6):
+        patterns.inav[0, 2].isig[_111_indexs[1][i], _111_indexs[0][i]] = 24.0
+    return patterns
+
+
+@pytest.fixture
+def test_lib_gen():
+    diff_gen = DiffractionGenerator(
+        accelerating_voltage=200,
+        precession_angle=1,
+        scattering_params=None,
```

Comment:
Do your tests pass locally? I think your tests fail in CI because you pass this parameter, which shouldn't be passed, I think?

Reply:
See for example https://github.com/pyxem/pyxem/runs/4802177150?check_suite_focus=true#step:8:227.

Reply:
They do pass locally.  I will check this.

Reply:
Yeah, this is the problem.  The tests run with the release version of diffsims which is old at the point and doesn't include None as an option. I assume I should rewrite the tests to work with that version?

Reply:
Always test against released versions!

Reply:
Great! Tests pass: https://github.com/pyxem/pyxem/runs/4802724206?check_suite_focus=true#step:8:72